smart_card_connector_app/src/background.js

```diff
@@ -44,37 +46,61 @@ const JS_LOGS_HANDLER_MESSAGE_TYPE = 'js_logs_handler';
  */
 var logger = GSC.Logging.getScopedLogger('ConnectorApp.BackgroundMain');
 
-// Used to forward logs collected on the JS side to the NaCl module's stdout, in
-// order to simplify accessing them in some configurations.
-// Note that this object needs to be created as early as possible, in order to
-// not miss any important log.
-const logBufferForwarderToNaclModule = new GSC.LogBufferForwarder(
-    GSC.Logging.getLogBuffer(), JS_LOGS_HANDLER_MESSAGE_TYPE);
+if (GSC.ExecutableModule.TOOLCHAIN === GSC.ExecutableModule.Toolchain.PNACL) {
+  // Used to forward logs collected on the JS side to the NaCl module's stdout,
+  // in order to simplify accessing them in some configurations.
+  // Note that this object needs to be created as early as possible, in order to
+  // not miss any important log.
+  var logBufferForwarderToNaclModule = new GSC.LogBufferForwarder(
+      GSC.Logging.getLogBuffer(), JS_LOGS_HANDLER_MESSAGE_TYPE);
+}
 
 const extensionManifest = chrome.runtime.getManifest();
 logger.info(
     `The Smart Card Connector app (id "${chrome.runtime.id}", version ` +
     `${extensionManifest.version}) background script started. Browser ` +
     `version: "${window.navigator.appVersion}".`);
 
-var naclModule = new GSC.NaclModule(
-    'nacl_module.nmf', GSC.NaclModule.Type.PNACL);
-naclModule.addOnDisposeCallback(naclModuleDisposedListener);
+/**
+ * Loads the binary executable module depending on the toolchain configuration.
+ * @return {!GSC.ExecutableModule}
+ */
+function createExecutableModule() {
+  switch (GSC.ExecutableModule.TOOLCHAIN) {
+    case GSC.ExecutableModule.Toolchain.PNACL:
+      return new GSC.NaclModule(
+          'nacl_module.nmf', GSC.NaclModule.Type.PNACL);
+    case GSC.ExecutableModule.Toolchain.EMSCRIPTEN:
+      return new GSC.EmscriptenModule('nacl_module');
+  }
+  GSC.Logging.fail(`Cannot load executable module: unknown toolchain ` +
+                   `${GSC.ExecutableModule.TOOLCHAIN}`);
+  goog.asserts.fail();
+}
 
-naclModule.getLoadPromise().then(() => {
-  // Skip forwarding logs that were received from the NaCl module or generated
-  // while sending messages to it, in order to avoid duplication and/or infinite
-  // recursion.
-  logBufferForwarderToNaclModule.ignoreLogger(
-      naclModule.logMessagesReceiver.logger.getName());
-  // Start forwarding all future log messages collected on the JS side, but also
-  // immediately post the messages that have been accumulated so far.
-  logBufferForwarderToNaclModule.startForwarding(
-      naclModule.getMessageChannel());
-}, () => {});
+/**
+ * The binary executable module that contains the actual smart card client code.
+ * @type {!GSC.ExecutableModule}
+ */
+const executableModule = createExecutableModule();
+executableModule.addOnDisposeCallback(naclModuleDisposedListener);
```

Comment:
nit: Rename "naclModuleDisposedListener" and change its message depending on TOOLCHAIN (unless this callback cannot happen for Wasm)